tests/std/tests/P1425R6_queue_stack_constructors/test.cpp

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <deque>
+#include <forward_list>
+#include <iostream>
+#include <memory>
+#include <queue>
+#include <set>
+#include <stack>
+#include <type_traits>
+#include <utility>
+#include <vector>
+using namespace std;
+
+constexpr int some_data[]       = {0, 1, 2, 3, 4, 5};
+constexpr int additional_data[] = {6, 7, 8, 9, 10, 11};
+
+template <typename T>
+struct custom_allocator {

```

Comment:
We additionally need to provide `operator==` accepting `custom_allocator`s of different types to satisfy the Allocator requirements. In C++14/17 we also need to provide `operator!=`.